libs/feature/search/src/lib/utils/links/link-utils.ts

```diff
@@ -1,67 +1,74 @@
-import { MetadataLink } from '@geonetwork-ui/util/shared'
+import { MetadataLinkValid } from '@geonetwork-ui/util/shared'
 import { marker } from '@biesbjerg/ngx-translate-extract-marker'
 
 marker('downloads.wfs.featuretype.not.found')
 
-export enum DownloadFormatType {
-  WFS = 'WFS',
-  FILE = 'FILE',
+const FORMATS = {
+  csv: ['csv'],
+  geojson: ['geojson'],
+  json: ['json'],
+  shp: ['shp', 'shape', 'zipped-shapefile'],
+  kml: ['kml', 'kmz'],
+  gpkg: ['gpkg', 'geopackage'],
+  excel: ['xls', 'xlsx', 'ms-excel', 'openxmlformats-officedocument'],
+  pdf: ['pdf'],
+  jpg: ['jpg', 'jpeg', 'jfif', 'pjpeg', 'pjp'],
+  zip: ['zip'],
 }
 
-export function getDownloadFormat(
-  link: MetadataLink,
-  type: DownloadFormatType
-): string {
+export function getWfsFormat(link: MetadataLinkValid): string {
+  for (const format in FORMATS) {
+    for (const alias of FORMATS[format]) {
+      if ('format' in link && new RegExp(`${alias}`, 'i').test(link.format))
+        return `WFS:${format}`
+    }
+  }
+  return undefined
+}
+
+export function getFileFormat(link: MetadataLinkValid): string {
+  if (link.format) {
+    return link.format
+  }
   if ('protocol' in link && /^WWW:DOWNLOAD/.test(link.protocol)) {
     // mime types in protocol
     const matches = link.protocol.match(/^WWW:DOWNLOAD:(.+\/.+)$/)
     if (matches !== null) {
-      const mimeType = matches[1]
-      switch (mimeType) {
-        case 'application/json':
-          return 'json'
-        case 'application/geo+json':
-        case 'application/vnd.geo+json':
-          return 'geojson'
-        case 'text/csv':
-        case 'application/csv':
-          return 'csv'
-        case 'x-gis/x-shapefile':
-          return 'shp'
-        case 'application/vnd.ms-excel':
-        case 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet':
-          return 'excel'
-      }
+      const mimeType = checkMimeType(matches[1])
```

Comment:
```suggestion
      return checkMimeType(matches[1])
```

Reply:
Ah thanks, not seeing the forest because of all the trees